shared/src/main/java/uk/gov/di/authentication/shared/services/TokenService.java

```diff
@@ -278,14 +278,43 @@ private SignedJWT generateIDToken(
         idTokenClaims.setClaim("vtm", trustMarkUri.toString());
 
         try {
-            return generateSignedJWT(
+            return generateSignedJwtUsingExternalKey(
                     idTokenClaims.toJWTClaimsSet(), Optional.empty(), signingAlgorithm);
         } catch (com.nimbusds.oauth2.sdk.ParseException e) {
             LOG.error("Error when trying to parse IDTokenClaims to JWTClaimSet", e);
             throw new RuntimeException(e);
         }
     }
 
+    public AccessToken generateStorageToken(Subject internalPairwiseSubject) {
+
+        LOG.info("Generating storage token");
+        Date expiryDate = NowHelper.nowPlus(configService.getSessionExpiry(), ChronoUnit.SECONDS);
+        var jwtID = UUID.randomUUID().toString();
+
+        LOG.info("Storage token being created with JWTID: {}", jwtID);
+
+        List<String> aud =
+                List.of(
+                        configService.getCredentialStoreURI().toString(),
+                        configService.getIPVAudience());
+
+        JWTClaimsSet.Builder claimSetBuilder =
+                new JWTClaimsSet.Builder()
+                        .issuer(configService.getFrontendBaseUrl())
```

Comment:
Let's go with this for now, but I wonder if this should be the auth frontend API domain instead? Let's start a converstation about this with Auth and VC Storage

Reply:
This one doesn't block merge